### PR TITLE
feat(examples): Adds tf template for bring your own vpc and eks

### PR DESCRIPTION
**Pull request by venky-wandb**
A few of customers in the past have requested to modify our terraform to allow provisioning resources into existing VPC and EKS clusters. This template helps with setting it up.

For context, customer who have asked for this in the past:
- DoorDash Labs
- Apple
- DIU

### CODE REVIEW

File: examples/byo-vpc-eks/main.tf [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  license = var.wandb_license
+
+  host                       = module.wandb_infra.url
+  bucket                     = "s3://${module.wandb_infra.bucket_name}"
+  bucket_aws_region          = module.wandb_infra.bucket_region
+  bucket_queue               = "internal://"
+  bucket_kms_key_arn         = module.wandb_infra.kms_key_arn
+  database_connection_string = "mysql://${module.wandb_infra.database_connection_string}"
+  # redis_connection_string    = "redis://${module.wandb_infra.elasticache_connection_string}?tls=true&ttlInSeconds=604800"
```
Comment (flamarion) [inline examples/byo-vpc-eks/main.tf:104]: Nitpicky: Better remove it if we will not use it. Customers will ask why we are not using it.

File: examples/byo-vpc-eks/main.tf [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  network_cidr = var.vpc_cidr
+
+  network_private_subnets          = var.network_private_subnets
+  network_public_subnets           = var.network_public_subnets
+  network_database_subnets         = var.network_database_subnets 
+  network_private_subnet_cidrs     = var.network_private_subnet_cidrs
+  network_public_subnet_cidrs      = var.network_public_subnet_cidrs
+  network_database_subnet_cidrs    = var.network_database_subnet_cidrs
+  # create_elasticache               = false
+  # network_elasticache_subnet_cidrs =
```
Comment (flamarion) [inline examples/byo-vpc-eks/main.tf:52]: Nitpicky: Better remove if we will not use it.

**Review by flamarion — APPROVED** [review]
I didn't test but it looks good.

File: examples/byo-vpc-eks/variables.tf [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    type = list[string]
+    description = "Subnet CIDRs"
+}
+
+variable network_database_subnet_cidrs {
+    type = list[string]
+    description = "Subnet CIDRs"
+}
+
+eks_cluster_version {
```
Comment (wandb-zacharyblasczyk) [inline examples/byo-vpc-eks/variables.tf:126]: I think something looks wrong here. Does this work?

Comment (venky-wandb) [reply examples/byo-vpc-eks/variables.tf]: Ahh, added this after running the terraform. It was missing variable keyword.

**Comment by jsbroks** [comment]
This PR is included in version 3.2.0 :tada:

**Pull request merged**